metaflow/plugins/argo/argo_workflows.py

```diff
@@ -883,6 +885,34 @@ def _visit(node, exit_node=None, templates=None, dag_tasks=None):
                         )
                     )
                 ]
+                # We need to add the split-index and root-input-path for the last step in any foreach to be able to generate
```

Comment:
Can you add more comments in the code about the need for `root-input-path`. Given the complexity of this work, it will be non-trivial to ship further changes without full context on this change.